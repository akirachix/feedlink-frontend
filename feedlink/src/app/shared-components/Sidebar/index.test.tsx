import { render, screen } from "@testing-library/react";
import Sidebar from "./index";
import { usePathname } from "next/navigation";

jest.mock("next/navigation", () => ({
  usePathname: jest.fn(),
}));

describe("Sidebar Component", () => {
  beforeEach(() => {
    (usePathname as jest.Mock).mockReset();
  });

  it("renders logo with correct alt text", () => {
    render(<Sidebar />);
    const logo = screen.getByAltText("FeedLink Logo");
    expect(logo).toBeInTheDocument();
  });

  it("renders all navigation links", () => {
    render(<Sidebar />);
    const links = screen.getAllByRole("link");
    expect(links.length).toBe(5); 
    expect(screen.getByText("Dashboard")).toBeInTheDocument();
    expect(screen.getByText("Inventory")).toBeInTheDocument();
    expect(screen.getByText("Waste Claims")).toBeInTheDocument();
    expect(screen.getByText("Orders")).toBeInTheDocument();
    expect(screen.getByText("Log Out")).toBeInTheDocument();
  });

  it("applies active styles to Dashboard when on /dashboard or subpath", () => {
    (usePathname as jest.Mock).mockReturnValue("/dashboard");

    render(<Sidebar />);
    const dashboardLink = screen.getByText("Dashboard").closest("a");

    expect(dashboardLink).toHaveClass("bg-white");
    expect(dashboardLink).toHaveClass("text-[var(--secondary-color)]");
    expect(dashboardLink).toHaveClass("font-semibold");
  });

  it("applies hover styles on non-active links (class names exist)", () => {
    (usePathname as jest.Mock).mockReturnValue("/inventory");

    render(<Sidebar />);
    const dashboardLink = screen.getByText("Dashboard").closest("a");

    expect(dashboardLink).not.toHaveClass("bg-white");
    expect(dashboardLink).not.toHaveClass("text-[var(--secondary-color)]");
    expect(dashboardLink).toHaveClass("hover:bg-white");
    expect(dashboardLink).toHaveClass("hover:text-[var(--secondary-color)]");
  });

  it("matches snapshot", () => {
    (usePathname as jest.Mock).mockReturnValue("/");

    const { asFragment } = render(<Sidebar />);
    expect(asFragment()).toMatchSnapshot();
  });
});