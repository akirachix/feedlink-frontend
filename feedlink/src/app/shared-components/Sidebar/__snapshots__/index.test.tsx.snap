// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Sidebar Component matches snapshot 1`] = `
<DocumentFragment>
  <aside
    class="w-72 min-h-screen relative flex flex-col"
    style="background-image: url("/images/sidebar.png"); background-size: cover; background-repeat: no-repeat; background-position: bottom; color: white;"
  >
    <div
      class="absolute top-16 left-8 flex flex-col items-start gap-2 select-none"
    >
      <img
        alt="FeedLink Logo"
        class="object-contain"
        height="70"
        src="/images/logo.svg"
        width="350"
      />
    </div>
    <nav
      class="flex flex-col gap-3 px-6 pt-40 pb-6 flex-grow"
    >
      <a
        class="flex items-center gap-3 rounded-md px-4 py-2 cursor-pointer transition-colors duration-200
                text-white hover:bg-white hover:text-[var(--secondary-color)]"
        href="/dashboard"
      >
        <svg
          fill="none"
          height="20"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="7"
            width="7"
            x="3"
            y="3"
          />
          <rect
            height="7"
            width="7"
            x="14"
            y="3"
          />
          <rect
            height="7"
            width="7"
            x="14"
            y="14"
          />
          <rect
            height="7"
            width="7"
            x="3"
            y="14"
          />
        </svg>
        <span
          class="text-lg"
        >
          Dashboard
        </span>
      </a>
      <a
        class="flex items-center gap-3 rounded-md px-4 py-2 cursor-pointer transition-colors duration-200
                text-white hover:bg-white hover:text-[var(--secondary-color)]"
        href="/inventory"
      >
        <svg
          fill="none"
          height="20"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 16V8a2 2 0 0 0-1-1.73l-7-4a2 2 0 0 0-2 0l-7 4A2 2 0 0 0 3 8v8a2 2 0 0 0 1 1.73l7 4a2 2 0 0 0 2 0l7-4A2 2 0 0 0 21 16z"
          />
          <polyline
            points="3.27 6.96 12 12.01 20.73 6.96"
          />
          <line
            x1="12"
            x2="12"
            y1="22.08"
            y2="12"
          />
        </svg>
        <span
          class="text-lg"
        >
          Inventory
        </span>
      </a>
      <a
        class="flex items-center gap-3 rounded-md px-4 py-2 cursor-pointer transition-colors duration-200
                text-white hover:bg-white hover:text-[var(--secondary-color)]"
        href="/wasteclaims"
      >
        <svg
          fill="none"
          height="20"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="3 6 5 6 21 6"
          />
          <path
            d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
          />
          <line
            x1="10"
            x2="10"
            y1="11"
            y2="17"
          />
          <line
            x1="14"
            x2="14"
            y1="11"
            y2="17"
          />
        </svg>
        <span
          class="text-lg"
        >
          Waste Claims
        </span>
      </a>
      <a
        class="flex items-center gap-3 rounded-md px-4 py-2 cursor-pointer transition-colors duration-200
                text-white hover:bg-white hover:text-[var(--secondary-color)]"
        href="/orders"
      >
        <svg
          fill="none"
          height="20"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="9"
            cy="21"
            r="1"
          />
          <circle
            cx="20"
            cy="21"
            r="1"
          />
          <path
            d="M1 1h4l2.68 13.39a2 2 0 0 0 2 1.61h9.72a2 2 0 0 0 2-1.61L23 6H6"
          />
        </svg>
        <span
          class="text-lg"
        >
          Orders
        </span>
      </a>
      <div
        class="mt-auto"
      />
      <a
        class="flex items-center gap-3 rounded-md px-4 py-2 text-white hover:bg-white hover:text-[var(--secondary-color)] transition-colors duration-200 mb-20"
        href="/logout"
      >
        <svg
          fill="none"
          height="20"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
          />
          <polyline
            points="16 17 21 12 16 7"
          />
          <line
            x1="21"
            x2="9"
            y1="12"
            y2="12"
          />
        </svg>
        <span
          class="text-lg"
        >
          Log Out
        </span>
      </a>
    </nav>
  </aside>
</DocumentFragment>
`;
